MonoGame.Framework/Graphics/Vertices/VertexBuffer.cs

```diff
@@ -26,10 +26,6 @@ protected VertexBuffer(GraphicsDevice graphicsDevice, VertexDeclaration vertexDe
             this.VertexCount = vertexCount;
             this.BufferUsage = bufferUsage;
 
-            // Make sure the graphics device is assigned in the vertex declaration.
-            if (vertexDeclaration.GraphicsDevice != graphicsDevice)
-                vertexDeclaration.GraphicsDevice = graphicsDevice;

```

Comment:
@KonajuGames - So that sounds exactly how we had things working here right?

Really the worst thing this could do would be keeping the old disposed `GraphicsDevice` class in memory.  Since it was disposed it should have already released all native resources so this reference shouldn't keep anything big in memory.

I think these two lines can be restored without any leaks appearing as long as we're disposing everything else correctly.